File: headless_solutions/complement_of_base_10_integer/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var bitwiseComplement = function(n) {
    if (n == 0) return 1;
    let X = 1;
    while (n > X) X = (X << 1) + 1;
    return X ^ n;
};

File: headless_solutions/complement_of_base_10_integer/solution.py
import collections
import functools
import itertools
import math
import heapq
import bisect
from typing import *
from collections import *
from functools import *
from heapq import *
from bisect import *

class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        X = 1
        while n > X:
            X = (X << 1) + 1
        return X ^ n
